tests/example_book/_build/html/notebooks/code_py.html

```diff
@@ -387,6 +387,34 @@ <h1>Code<a class="headerlink" href="#code" title="Permalink to this headline">#<
 </div>
 </div>
 </div>
+<p>Next, here’s a code block that ought to end up as a titled example (<span class="xref std std-ref">hello_tim</span>).</p>
+<div class="cell tag_example docutils container">
+<div class="cell_input docutils container">
+<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># hello_tim</span>
+<span class="c1"># An example &quot;Hello Tim&quot; code example</span>
+
+<span class="k">def</span> <span class="nf">hello</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;Tim&quot;</span><span class="p">):</span>
+    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Hello, </span><span class="si">{</span><span class="n">name</span><span class="si">}</span><span class="s2">! Nice to meet you!&quot;</span><span class="p">)</span>
+</pre></div>
+</div>
+</div>
+</div>
+<p>Finally, here’s a titled example that has output. See <span class="xref myst">hello_tim_with_output</span>.</p>
```

Comment:
It's an outstanding question about whether or not the output should be included in the example. In general we don't encourage output in examples AFAIK, but suffice to say right now the output would just dangle. I am hoping that we can discuss here if we need to support output for examples or not. If we do, I'll add a commit to this PR to do so, but my gut says we probably don't want to encourage this.

Reply:
In my experience with authors, at some point one will want to include output even if we discourage it. I'd lean toward including things where possible. Hopefully not a lot of extra work :)

Reply:
This turned out to be fairly easy in the end! See updated commit.